src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectGrantController.java

```diff
@@ -91,14 +99,26 @@ public CatalogObjectGrantMetadata createCatalogObjectGrantForAUser(
             }
             // Check Grants
             if (!restApiAccessService.isBucketAccessibleByUser(sessionIdRequired, sessionId, bucketName)) {
-                if (!catalogObjectGrantService.checkInCatalogObjectGrantsIfTheUserOrUserGroupHasAdminRightsOverTheCatalogObject(user,
-                                                                                                                                bucketName,
-                                                                                                                                catalogObjectName)) {
-                    throw new CatalogObjectGrantAccessException(bucketName, catalogObjectName);
+                List<CatalogObjectGrantMetadata> grants = catalogObjectGrantService.getAllAssignedCatalogObjectGrantsForUser(user)
```

Comment:
same remark as above these stream filter collect are difficult to read. Refactor using a private method with understandable name